Socket-programing-build

flatroad opened the pull request:
<!-- 
📝 PR 작성 가이드
- 제목: [타입] 간단한 요약 (예: [feat] 사용자 로그인 기능 추가)
- 각 섹션을 빠뜨리지 말고 작성해주세요
- 불필요한 주석은 삭제하고 제출해주세요
-->

## 📋 요약 (Summary)
<!-- 작업 내용을 한 줄로 요약해주세요 -->
<!-- 예: 사용자 인증 시스템에 소셜 로그인 기능을 추가했습니다. -->
winsock을 사용하기 위한 빌드 처리.

## 🔧 주요 구현(변경)사항
- log 출력 유형에 따른 매크로 함수 추가.
- 메모리 할당/관리를 체크할 수 있는 매크로 함수 추가.
- WSAStartup, WSACleanup를 통해 WSADATA 초기화/해제.

### 📝 상세 내용 (Describe your changes)
<!-- 변경 사항에 대한 자세한 설명을 작성해주세요 -->
<!-- 
예시:
- Google OAuth 2.0을 사용한 소셜 로그인 구현
- 기존 로그인 페이지에 소셜 로그인 버튼 추가  
- 사용자 정보 연동 로직 구현
-->
- log의 유형을 구분해 줄수 있게 enum class 사용.
- log 유형에 따라 메세지 출력 방식 설정.
- log 출력을 편리하게 하기 위해 매크로 함수 사용.
- ms에서 제공하는 메모리 leak 체크 방식 사용.
- new 함수 재정의를 통해 디버깅 강화.
- 메모리 관련 함수 매크로 사용으로 사용성 강화.
- WSAStartup를 통해 winsock 사용에 필요한 운영체제, dll 할당 및 초기화.
- WSACleanup을 통해 winsock에 사용되었던 운영체제, dll 해제.

### 📱 스크린샷 (해당하는 경우)
<!-- UI 변경이 있는 경우 스크린샷을 첨부해주세요 -->
<img width="854" alt="image" src="https://github.com/user-attachments/assets/dc03630a-d64a-4077-b74f-ec5255b297a6" />


## 🔗 Issue Number or Link
<!-- 관련된 이슈가 있다면 연결해주세요 -->
- Refs #1

## ✅ 체크리스트
<!-- PR 제출 전 확인해주세요 -->
- [X] 코드 리뷰 요청 완료
- [X] 브랜치가 최신 상태로 업데이트됨
- [X] 커밋 메시지가 명확하고 일관됨

## 📋 추가 정보
<!-- 리뷰어가 알아야 할 추가 정보가 있다면 작성해주세요 -->
<!-- 예: 특별한 주의사항, 배포 시 고려사항 등 -->


## 👥 리뷰어
<!-- 리뷰를 요청할 사람들을 멘션해주세요 -->
@JSG0626 

---
<!-- 
🎉 PR 작성 완료!
리뷰어가 쉽게 이해할 수 있도록 명확하고 자세하게 작성해주세요.
-->

flatroad committed:
feat(SocketIniter): socket-programing-build

소켓 프로그래밍을 위한 winsock 초기화
- memorydebug를 위한 매크로 추가
- log 출력을 위한 매크로 추가

Refs: #1

flatroad committed:
feat(SocketIniter): socket-programing-build

소켓 프로그래밍을 위한 winsock 초기화
- memorydebug를 위한 매크로 추가
- log 출력을 위한 매크로 추가

Refs: #1

JSG0626 commented:
프로그램의 기본 베이스를 탄탄히 잡고 가려는게 보여서 앞으로 디버그 및 확장에 용이할 것으로 보입니다. 특히 로그 관련 매크로와 시스템이 잘 구현이 되어있어서 코드가 깔끔하고 체계적으로 보입니다. 인코딩 관련 부분만 신경쓰고 올리시면 될 것 같습니다. 승인하도록 하겠습니다.

JSG0626 reviewed (approved):
프로그램의 기본 베이스를 탄탄히 잡고 가려는게 보여서 앞으로 디버그 및 확장에 용이할 것으로 보입니다. 특히 로그 관련 매크로와 시스템이 잘 구현이 되어있어서 코드가 깔끔하고 체계적으로 보입니다. 인코딩 관련 부분만 신경쓰고 올리시면 될 것 같습니다. 승인하도록 하겠습니다.

flatroad committed:
refactor(SocketIniter): 코드 리뷰에 따라 파일 인코딩 변경 및 구조 개선.

1. 파일 인코딩을 변경했습니다. (한글 깨짐현상 해결)
2. 일부 함수 설명 주석을 추가했습니다.
3. init 함수 리턴 값을 enum class로 수정했습니다.
4. main문을 개선했습니다.

Refs: #1, #3

flatroad commented:
@JSG0626 

코드 리뷰 감사합니다. 리뷰에서 지적하신 사항 (인코딩) 은 수정했습니다.
감사합니다.
